### PR TITLE
Add missing argument to `open` call

### PR DESCRIPTION
**Pull request by robertoaloi**
_TL;DR;_ When using the O_CREAT flag in the 'open' function call, you'd better to provide the mode explicitly, or the Ubuntu user cries.

When trying to compile nsca on my Ubuntu 14.04, I got the following:

```
/usr/include/x86_64-linux-gnu/bits/fcntl2.h:50:24: error: call to ‘__open_missing_mode’ declared with attribute error: open with O_CREAT in second argument needs 3 arguments
    __open_missing_mode ();
                        ^
make[1]: *** [nsca] Error 1
```

Starting from Ubuntu 8.10, the compiler flag -D_FORTIFY_SOURCE=2 is used by default. This causes the compiler to check for compile-time best-practices errors when calling certain libc functions.
This is the case for `open`, where the mode should be explicitly set rather than inferred from the process umask.

References:

http://linux.die.net/man/2/open
https://wiki.ubuntu.com/ToolChain/CompilerFlags?action=show&redirect=CompilerFlags#A-D_FORTIFY_SOURCE.3D2

### CODE REVIEW

**Comment by DennisonWilliams** [comment]
I confirmed this patch works on Ubuntu 12.04

**Comment by jfrickson** [comment]
It looks like this was fixed in October 2015 via commit https://github.com/NagiosEnterprises/nsca/commit/b9f2c08dfdf31430f972a48781fc115c4af13cfe. Fix was based on a patch submitted in the tracker issue.
